creator-node/src/utils/legacyUtils.js

```diff
@@ -119,74 +126,113 @@ export async function findCIDInNetwork(
 
-  // generate signature
+  // Generate signature to auth fetching files
   const delegateWallet = config.get('delegateOwnerWallet').toLowerCase()
   const { signature, timestamp } = generateTimestampAndSignature(
     { filePath, delegateWallet },
     config.get('delegatePrivateKey')
   )
-  let node
 
-  for (let index = 0; index < creatorNodesFiltered.length; index++) {
-    node = creatorNodesFiltered[index]
+  let found = false
+  for (const { endpoint } of creatorNodesFiltered) {
     try {
-      const resp = await axios({
-        method: 'get',
-        url: `${node.endpoint}/file_lookup`,
-        params: {
-          filePath,
-          timestamp,
-          delegateWallet,
-          signature,
-          trackId
-        },
-        responseType: 'stream',
-        timeout: 1000
-      })
-      if (resp.data) {
-        await writeStreamToFileSystem(resp.data, filePath, /* createDir */ true)
+      found = await asyncRetry({
+        asyncFn: async (bail) => {
+          let response
+          try {
+            response = await axios({
+              method: 'get',
+              url: `${endpoint}/file_lookup`,
+              params: {
+                filePath,
+                timestamp,
+                delegateWallet,
+                signature,
+                trackId
+              },
+              responseType: 'stream',
+              timeout: 1000
+            })
+          } catch (e) {
+            if (
+              e.response?.status === 403 || // delist
+              e.response?.status === 401 || // unauth
+              e.response?.status === 400 || // bad req
+              e.response?.status === 404 // not found
+            ) {
+              bail(
+                new Error(
+                  `Content multihash=${cid} not available on ${endpoint} with statusCode=${e.response?.status}`
+                )
+              )
+              return
+            }
+
+            throw new Error(
+              `Failed to fetch content multihash=${cid} with statusCode=${e.response?.status}. Retrying..`
+            )
+          }
+
+          if (!response || !response.data) {
+            throw new Error('Received empty response from file lookup')
+          }
+
+          await writeStreamToFileSystem(
+            response.data,
+            filePath,
+            /* createDir */ true
+          )
 
-        // Verify that the file written matches the hash expected
-        const expectedCID = await LibsUtils.fileHasher.generateNonImageCid(
-          filePath
-        )
+          const isCIDProper = await verifyCIDIsProper({
+            cid,
+            path: filePath,
+            logger
+          })
+
+          if (!isCIDProper) {
+            try {
+              await fs.unlink(filePath)
+            } catch (e) {
+              logger.error(`Could not remove file at path=${path}`)
+            }
+
+            bail(new Error(`CID=${cid} from endpoint=${endpoint} is improper`))
+            return
```

Comment:
should these be `return false` here and above?

Reply:
the bail and then return is just to not retry (some weird async-retry format). it'll throw into the catch block later with the bailed error, and then return found=false since it was never set. 

i'll make this  line 232 be `found=true` instead of `return true` to only return the found value

Reply:
ah yep dope